### PR TITLE
Clean up flutter_gallery__transition_perf

### PR DESCRIPTION
**Pull request by CareF**
## Description

This is follow up of #62064

This PR tries to clean up `flutter_gallery__transition_perf` related with: 
- using the same app file for the driver test as #61621
- combine `runDemos` for e2e and create that for hybrid test. 

## Related Issues

N/A

## Tests

The PR itself is to update tests. 

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: dev/integration_tests/flutter_gallery/test_driver/transitions_perf_test.dart [hunk 1]
```diff
@@ -205,8 +211,12 @@ void main([List<String> args = const <String>[]]) {
           histogramPath);
 
       // Execute the remaining tests.
-      final Set<String> unprofiledDemos = Set<String>.from(_allDemos)..removeAll(kProfiledDemos);
-      await runDemos(unprofiledDemos.toList(), driver);
+      if (hybrid) {
```
[the remark this answers is not in the document]
Comment (CareF) [reply dev/integration_tests/flutter_gallery/test_driver/transitions_perf_test.dart:213]: Done.

File: dev/integration_tests/flutter_gallery/test_driver/transitions_perf_test.dart [hunk 2]
```diff
@@ -186,7 +188,11 @@ void main([List<String> args = const <String>[]]) {
       // Collect timeline data for just a limited set of demos to avoid OOMs.
       final Timeline timeline = await driver.traceAction(
         () async {
-          await runDemos(kProfiledDemos, driver);
+          if (hybrid) {
```
[the remark this answers is not in the document]
Comment (CareF) [reply dev/integration_tests/flutter_gallery/test_driver/transitions_perf_test.dart:190]: Done.

File: dev/devicelab/manifest.yaml [hunk 3]
```diff
@@ -762,6 +762,14 @@ tasks:
     required_agent_capabilities: ["linux/android"]
     flaky: true
 
+  flutter_gallery__transition_perf_hybrid:
+    description: >
+      Measures the performance of screen transitions in Flutter Gallery on
+      Android with hybrid of host-driven and self-driven.
```
[the remark this answers is not in the document]
Comment (CareF) [reply dev/devicelab/manifest.yaml]: Done.

File: dev/devicelab/lib/tasks/gallery.dart [hunk 4]
```diff
@@ -53,18 +63,18 @@ class GalleryTransitionTest {
     await inDirectory<void>(galleryDirectory, () async {
       await flutter('packages', options: <String>['get']);
 
-      final String testDriver = semanticsEnabled
-          ? '${testFile}_with_semantics.dart'
-          : '$testFile.dart';
+      driverFile ??= semanticsEnabled
```
[the remark this answers is not in the document]
Comment (CareF) [reply dev/devicelab/lib/tasks/gallery.dart]: Done

File: dev/integration_tests/flutter_gallery/test_driver/transitions_perf.dart [hunk 5]
```diff
@@ -7,18 +7,45 @@ import 'dart:convert' show JsonEncoder;
...
 import 'package:flutter_driver/driver_extension.dart';
 import 'package:flutter_gallery/gallery/demos.dart';
+import 'package:flutter_gallery/demo_lists.dart';
 import 'package:flutter_gallery/gallery/app.dart' show GalleryApp;
+import 'package:flutter_test/flutter_test.dart';
 import 'package:flutter/material.dart';
 
+import 'run_demos.dart';
+
+LiveWidgetController controller;
```
[the remark this answers is not in the document]
Comment (CareF) [reply dev/integration_tests/flutter_gallery/test_driver/transitions_perf.dart]: I made a class with `call` method to implement this, with single call of `Controller` construction. PTAL.

**Review by liyuqian — APPROVED** [review]
LGTM

**Pull request merged**